### DISCUSSION TITLE
Intent Classification

### DISCUSSION THREAD

**Discussion by pythiantech** [567-labs/instructor#430] [General]
@jxnl Firstly thank you so much for building something as cool as this!

I work in the maritime domain and was wondering if something like this is even possible:

Based on what a user writes, I want to be able to extract his intention and use that as an input parameter for an API call. For the sake of simplicity, let us assume that there are certain `activity types` that the API responds to. As an example, these could be:
 - SLOW_SPEED
 - ANCHORING
 - FISHING
 - RESEARCH
 - COURSE_DEVIATION

Now if a user writes something like, `Show me all vessels that sailed below 3 knots in the Arabian Sea`, I want to classify the intent as `SLOW_SPEED`. Similarly, a user input like `How many tankers were anchored in the port waiting area in the past one week` would translate to `ANCHORING` as the activity. 

Is this something that can be achieved using pydantic and instructor?

**Comment by jxnl** [comment]
https://jxnl.github.io/instructor/examples/batch_classification/

https://jxnl.github.io/instructor/examples/classification/

https://jxnl.github.io/instructor/blog/2023/09/17/rag-is-more-than-just-embedding-search/#case-study-1-metaphor-systems

**Reply by pythiantech** [reply]
Nice! I think this should solve my problem
